From cf8b8f58c02f3f2aa0cd19ca130992c465c2d9a3 Mon Sep 17 00:00:00 2001
From: xianhaox <xianhaox.qing@intel.com>
Date: Wed, 8 Nov 2017 18:43:44 +0800
Subject: [PATCH 02/12] Make TYPE_DISPLAY_OVERLAY to be trusted

The Hvac app used TYPE_DISPLAY_OVERLAY window type. It covers
almost the screen. Without trusting it will cause touch
event tapjacking security checking failure.

The TYPE_DISPLAY_OVERLAY window type can only be created by
system app. It's okay to trust this type of window.

Change-Id: I729fbb024c74ccfe25533499e7889eb5e548ec26
Tracked-On: https://jira01.devtools.intel.com/browse/OAM-63643
Signed-off-by: xianhaox <xianhaox.qing@intel.com>
Signed-off-by: Walter, Yan <walterx.yan@intel.com>
---
 services/inputflinger/InputWindow.cpp | 1 +
 services/inputflinger/InputWindow.h   | 1 +
 2 files changed, 2 insertions(+)

diff --git a/services/inputflinger/InputWindow.cpp b/services/inputflinger/InputWindow.cpp
index 3ae7972779..c43f42e2f0 100644
--- a/services/inputflinger/InputWindow.cpp
+++ b/services/inputflinger/InputWindow.cpp
@@ -50,6 +50,7 @@ bool InputWindowInfo::isTrustedOverlay() const {
             || layoutParamsType == TYPE_SECURE_SYSTEM_OVERLAY
             || layoutParamsType == TYPE_DOCK_DIVIDER
             || layoutParamsType == TYPE_ACCESSIBILITY_OVERLAY
+            || layoutParamsType == TYPE_DISPLAY_OVERLAY
             || layoutParamsType == TYPE_INPUT_CONSUMER;
 }
 
diff --git a/services/inputflinger/InputWindow.h b/services/inputflinger/InputWindow.h
index 5a48375910..6dbfff593f 100644
--- a/services/inputflinger/InputWindow.h
+++ b/services/inputflinger/InputWindow.h
@@ -102,6 +102,7 @@ struct InputWindowInfo {
         TYPE_BOOT_PROGRESS = FIRST_SYSTEM_WINDOW+21,
         TYPE_INPUT_CONSUMER = FIRST_SYSTEM_WINDOW+22,
         TYPE_NAVIGATION_BAR_PANEL = FIRST_SYSTEM_WINDOW+24,
+        TYPE_DISPLAY_OVERLAY = FIRST_SYSTEM_WINDOW+26,
         TYPE_MAGNIFICATION_OVERLAY = FIRST_SYSTEM_WINDOW+27,
         TYPE_ACCESSIBILITY_OVERLAY = FIRST_SYSTEM_WINDOW+32,
         TYPE_DOCK_DIVIDER = FIRST_SYSTEM_WINDOW+34,
-- 
2.17.1

